### ISSUE TITLE
Add more "hidden options" to the Advanced NVDA Settings

### ISSUE THREAD

**Issue by DrSooom** [nvaccess/nvda#9206]
### Actual behaviour:
The following "hidden options" can only be changed directly in the "nvda.ini":
+ beepSpeechModePitch
+ speechPercentageInterval
+ beepPercentageInterval
+ beepMinHZ
+ audioCoordinates_blurFactor
+ audioCoordinates_minVolume
+ audioCoordinates_maxVolume
+ audioCoordinates_minPitch
+ audioCoordinates_maxPitch
+ [UIA] minWindowsVersion
+ [UIA] enabled
### Expected behaviour:
1. Include them all to the Advanced NVDA Settings, so that they can be changed via the GUI.
2. Add a validation check for all of them too.
  + beepSpeechModePitch: text field (min = 55, max = 8372, default = 880)
Note: I guess the default value here is/was 8000 Hz, which hurts extremely in my ears. So reducing it directly to 880 Hz would be nice.
  + speechPercentageInterval: text field (min = 1, max = 25, default = 10)
  + beepPercentageInterval: text field (min = 1, max = 25, default = 1)
  + beepMinHZ: text field (min = 55, max = 4186, default = 110)
  + audioCoordinates_blurFactor: text field (min = 1, max = 15, default = 3)
  + audioCoordinates_minVolume: text field (min = 0.000, max = 0.999, default = 0.1)
  + audioCoordinates_maxVolume: text field (min = 0.001, max = 1.0, default = 1.0)
  + audioCoordinates_minPitch: text field (min = 55, max = 4186, default = 220)
  + audioCoordinates_maxPitch: text field (min = 110, max = 8372, default = 880)
  + [UIA] minWindowsVersion: text field (min = 6.1, max = 10.0, default = 6.1)
  + [UIA] enabled: checkbox (true and false, default = true)
3. Describe them all in detail in the NVDA's User Guide (like in chapter 15.1 "Command Line Options").
#### Note regarding the Hz values:
55 Hz = A1, 110 Hz = A2, 220 Hz = A3, 880 Hz = A5, 4186 Hz = C8 and 8372 Hz = C9
#### NVDA version:
+ The above mentioned "hidden options" are part from the "nvda.ini" of NVDA 2012.2.1 (and a little bit higher), as far as I remember me correctly. They were never deleted during updating NVDA from 2012.2.1/2013.2 to 2018.1. (I used NVDA 2012.2.1 to 2013.2 as a portable app.)
+ In newer NVDA versions, maybe since introducing the second schema version for the "nvda.ini", only different values in compare with the default once are saved in the "nvda.ini".
+ The "Advanced Settings" is a new category in the NVDA Settings in NVDA alpha-16619 (alpha build for the upcoming 2019.1), which I haven't tested now. I got this information from PR #9200.
#### Windows version:
Windows 7 to Windows 10 1809 (and 19H1 and higher)

**Comment by josephsl** [comment]
Hi, one major correction: UIA min Windows version key is gone (I removed it last year). Besides that, I have no comments on this proposal unless asked to comment on certain things. CC @MichaelDCurran, @JCSTeh

**Comment by Brian1Gaff** [comment]
Is any of this  really really top priority though given the current moves to 
python 3 and what that will mean to add ons and other things iin nvda.
 Several people would like some of the longer running issues to be fixed I 
think before adding complexity to the settings panels.
 Just a thought.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by leonardder** [comment]
#9200 intends to implement the most important advanced settings. I think this issue is good for a new developer whenever #9200 is merged into master. I'm pretty sure that it is very low on the priority list of both NV Access and other volunteer developers though.

**Comment by Adriani90** [comment]
At least for the speech and Audio Settings it would be good to have These two:
• speechPercentageInterval
• beepPercentageInterval
Maybe as part of speech Settings. We could rename the cathegory like "speech and Audio Settings".

Concerning the hz and pitch of tones, I would rather prefer replacing the tones with softer tones which are not so shrill.

**Comment by DrSooom** [comment]
@feerrenrut: I fully agree with you. But making them accessible via the GUI should be done first. Well, as everybody here will also acknowledge to, is that the priority of this issue here is just a p4. Personally I don't want to have hidden options, which aren't documented.

@Adriani90: See issue #8711. Your suggestion would split the progress bar audio feedback and their intervals into two different panels, which isn't user-friendly. Furthermore all settings in the speech panels are related to the current TTS, as far as I know and understood it correctly. So the two interval settings shouldn't be related to a TTS, because they are generally used.

**Comment by DrSooom** [comment]
Related issue: #6788

**Comment by zahra21** [comment]
hello.
can someone please tell me how can i increase caret move timeout in nvda 2017.3?
in issue 9200, i realized that this feature is available in gGUi since 2019.1 of nvda.
however, i did not find documentation how can i increase it in my favorite version which supports xp (2017.3)!
thanks for any of your help, God bless you all, and His infinite mercy i pray for nvda developers, programmers and supporters.
if my request is off topic, please answer my question via my email address.
nasrinkhaksar3@gmail.com

**Comment by feerrenrut** [comment]
Most of these config items aren't very user friendly, which is an important consideration. Rather than exposing these via the settings menu, I'd prefer to incorporate them into a new type of "audio theme" addon along with customisation of other sounds in NVDA.

**Comment by seanbudd** [comment]
#10804 includes some important discussion on how some of these could be implemented

**Comment by seanbudd** [comment]
Closing as won't accept.

- Many of these parameters have been removed.
- We intentionally don't surface most of these
- Technical users can modify them manually, and create add-ons to do it.
